cli/src/test/java/com/devonfw/tools/ide/io/FileAccessImplTest.java

```diff
@@ -0,0 +1,227 @@
+package com.devonfw.tools.ide.io;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.function.Function;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.context.IdeTestContextMock;
+
+public class FileAccessImplTest extends Assertions {
+
+  @Test
+  void testSymlink(@TempDir Path tempDir) {
+
+    IdeContext context = IdeTestContextMock.get();
+    FileAccess fileAccess = new FileAccessImpl(context);
+    Path dir = tempDir.resolve("dir");
+    fileAccess.mkdirs(dir);
+
+    Path file = tempDir.resolve("file");
+    try {
+      Files.write(file, "Hello World!".getBytes());
+    } catch (IOException e) {
+      throw new RuntimeException("When preparing testSymlink writing of " + file + " failed. ", e);
+    }
```

Comment:
You could manually put a file in the test resource folder to avoid these lines of creating writing a file:

Reply:
After discussing this with you and the team I changed this a bit but still act inside a tmp dir.